content/en/blog/_posts/kubernetes-1-27-efficient-selinux-relabeling-beta.md

```diff
@@ -0,0 +1,126 @@
+---
+layout: blog
+title: "Kubernetes 1.27: Efficient SELinux volume relabeling (Beta)"
+date: 2023-04-11T10:00:00-08:00
+slug: kubernetes-1-27-efficient-selinux-relabeling-beta
+---
+
+**Author:** Jan Šafránek (Red Hat)
+
+# The problem
+
+On Linux with Security-Enhanced Linux (SELinux) enabled, it's traditionally
+the container runtime who applies SELinux labels to a Pod and all it's volumes.
+Kubernetes only provides the SELinux label from Pod's Security Context fields
+to the container runtime.
+
+The container runtime then recursively changes SELinux label on all files that
+are visible to the Pod's containers. This can be time-consuming, if there are
+many files on the volume, especially when the volume is on a remote filesystem.
+
+{{< note >}}
+If a container uses `subPath` of a volume, only that `subPath` of the whole
+volume is relabeled. This allows two pods that have two different SELinux labels
+to use the same volume, as long as they use different subpaths of it.
+{{< /note >}}
+
+{{< note >}}
+If a Pod does not have any SELinux label assigned in Kubernetes API, the
+container runtime assigns a unique random one, so a process that potentially
+escapes the container boundary cannot access data of any other container on the
+host. The container runtime still recursively relabels all pod volumes with this
+random SELinux label.
+{{< /note >}}
+
+# Improvement using mount options
+
+If a Pod + its volume satisfies **all** following conditions, Kubernetes will
```

Comment:
The conditions to be met are related to OS the pod is being scheduled on and cluster configuration (feature gates) - technically there is no condition that a volume needs to meet, right? Could be rewritten as:

If certain conditions are met, Kubernetes will mount a volume directly with the right SELinux label, which takes constant time, instead of container runtime having to recursively relabel any files on it. These conditions include:

1. The operating system which a Pod is scheduled on must support SELinux.
...
...